source/agora/stats/StatsServer.d

```diff
@@ -0,0 +1,43 @@
+module agora.stats.StatsServer;
+
+import ocean.util.prometheus.collector.CollectorRegistry;
+import vibe.vibe;
+
+import std.stdio;
+
+class StatsServer
+{
+    private HTTPListener http_listener;
```

Comment:
Must of this complication can be avoided by using the REST generator and an API.

E.g.
```D
interface PrometheusAPI
{
    string getMetrics();
}

class FullNode : FullNodeAPI, PrometheusAPI {}
```

Then it just fits into LocalRest. We can also do `registerRestInterface!(PrometheusAPI)(node)` to expose *only* the Prometheus API to Vibe.d's REST generator, and handle the rest ourselves.

Reply:
we can do that but the real networking code in the StatsServer is very simple:
```D
        auto router = new URLRouter;
        router.get("/metrics", &handle_metrics);

        auto settings = new HTTPServerSettings;
        settings.port = port;
        http_listener = listenHTTP(settings, router);
```

and

```D
 private void handle_metrics (HTTPServerRequest req, HTTPServerResponse res)
 {
     res.writeBody(cast(const(ubyte[])) getCollectorRegistry().collect(),"text/plain");
 }
```
that would be the only thing Rest would help us to avoid

Reply:
Well not the only thing, because then you can call this endpoint from LocalRest for free, no need to version it out / do dependency injection.

Reply:
Discussed via Slack, and unfortunately the REST generator is too limited for now.